src/Architectures.jl

```diff
@@ -31,17 +32,26 @@ Run Oceananigans on a single NVIDIA CUDA GPU.
 """
 struct GPU <: AbstractArchitecture end
 
+"""
+    MetalBackend <: AbstractArchitecture
+
+Run Oceananigans on a single M1 GPU.
+"""
+struct MetalBackend <: AbstractArchitecture end
```

Comment:
```suggestion
struct Metal <: AbstractArchitecture end
```

Reply:
as discussed elsewhere, the meaning of "backend" is unclear. We use "backend" to refer to how data is stored for `FieldTimeSeries` but we haven't used that word for architectures. I think actually just "Metal" is better here. "MetalGPU" could also work.

When we worked on AMDGPU support, we implemented this differently: it was something like

```julia
struct GPU{D} <: AbstractArchitecture
    device :: D
end
```

Then if you asked for a GPU you could either speicfy that you wanted one in particular; otherwise we would try to figure out which GPU was present and use that one (either AMD or CUDA).

Reply:
I initially didn't call it just `Metal` because that conflicts with the package name

Reply:
After thinking about this more, I feel that generalizing the `GPU` struct is the right approach. The concept is the same as for any "other" GPU (Nvidia, AMD, Intel). If we alternatively use `MetalGPU` then I think we'd also have to change `GPU` to `CUDAGPU`, which I don't like so much. What do you think?

Reply:
Yeah this seems reasonable, although is there anything that is specific to GPUs rather than CPUs that makes a Metal/CUDA subtype of GPU different from CPU? I guess just from a naming perspective it makes more sense though as almost all use cases someone will have both a CPU and one type of GPU.